File: Front-End/lib/data/DTOs/update_user_dto.dart
class UpdateUserDto {
  final String? fullName;
  final String? phone;
  final String? bio;
  final String? newPassword;
  final String? oldPassword;

  UpdateUserDto({
    this.fullName,
    this.phone,
    this.bio,
    this.newPassword,
    this.oldPassword
  });

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = {};

    if (fullName != null) data['fullName'] = fullName;
    if (phone != null) data['phone'] = phone;
    if (bio != null) data['bio'] = bio;

    if (newPassword != null || oldPassword != null){
      data['oldPassword'] = oldPassword;
      data['newPassword'] = newPassword;
    }

    return data;
  }
}

File: API/LocalDBApi/LocalDBApi/DTOs/UpdateUserDto.cs
namespace LocalDBApi.DTOs
{
    public class UpdateUserDto
    {
        public string? FullName { get; set; }
        public string? Phone { get; set; }
        public string? Bio { get; set; }
        public string? NewPassword { get; set; }
        public string? OldPassword { get; set; }
    }
}
